optimum/exporters/onnx/base.py

```diff
@@ -558,3 +558,104 @@ def get_decoder_onnx_config(
             f"{config.model_type} decoder export is not supported yet. ",
             f"If you want to support {config.model_type} please propose a PR or open up an issue.",
         )
+
+
+class OnnxConfigWithLoss(OnnxConfig, ABC):
```

Comment:
Created only a single wrapper class to for training. Earlier there were 3 configs for `OnnxConfigWithLoss`, `OnnxConfigWithPastAndLoss`, `OnnxSeq2SeqConfigWithPastAndLoss`. The single config should be able to handle all the above use-cases. 

@JingyaHuang  to comment if we need them separately.

Reply:
Yeah one wrapper in great!

Reply:
Actually, with_loss configs were temporary, it is necessary for the `evaluate` in the trainer, and by the end of the day we should have a more generic wrapper to allow users to export all possible outputs as in transformers. This is something that we can tackle in upcoming PRs.

Reply:
@michaelbenayoun @fxmarty

Reply:
Maybe a small nit, but should we put it in another file?
This file is like for base classes, and this one seems to be ready to go, it would make more sense to put it in `model_configs.py` for instance, but even there, I think it would be out of place... So maybe a new file?

WDYT @JingyaHuang @mht-sharma ?

Reply:
Do not have a strong opinion on this.  Could move this config in a new file `training_config.py`?